### ISSUE TITLE
About the RANSAC 

### ISSUE THREAD

**Issue by atztao** [magicleap/SuperGluePretrainedNetwork#56]
I think it's great job. But i have a problem, i test the pretrained model, i foud most time the output not have lots of erro match, so you used the RANSAC in the model to do it or others way to remove the erro matches ?

**Comment by Skydes** [comment]
Please refer to the paper and to the code. We use RANSAC to find the geometric inliers and estimate the transformation.

**Comment by atztao** [comment]
> Please refer to the paper and to the code. We use RANSAC to find the geometric inliers and estimate the transformation.

Thanks to your reply. And how do you set the unmatch and match keypoints for the inputs, that's balance?  Also, how long will it take to converge? 2k or 20k?

**Comment by Skydes** [comment]
Sorry, I don't understand your questions.

**Comment by atztao** [comment]
Oh, sorry. I mean that if i used the the num of 1024 keypoints for input, the match pairs should have 512 or more ?

**Comment by Skydes** [comment]
SuperGlue accepts an arbitrary number of keypoints in each image of the pair.

**Comment by atztao** [comment]
Hey, i train on the SIFT used bs=8 but is not good performance or it not work, but when i used bs =1 is good, i trained 10k.

**Comment by Skydes** [comment]
That is quite surprising: larger batches should give better performance. I also don't expect only 10k iterations to be sufficient for good performance.

**Comment by atztao** [comment]
The input of num of the match should be equal the num of the unmatch?

**Comment by Skydes** [comment]
I am not sure to understand your question.

**Comment by atztao** [comment]
Hh, i mean the match pair vs the unmatch pair should be equal?

**Comment by Skydes** [comment]
No, these numbers can be different.

**Comment by atztao** [comment]
Finally, this is a good job, thank you. But i have another questions, that the network is easy to get 'NAN' loss, if not add the bn or relu in the last layer.

**Comment by atztao** [comment]
Finally, this is a good job, thank you. But i have another questions, that the network is easy to get 'NAN' loss, if not add the bn or relu in the last layer.

**Comment by Skydes** [comment]
I this might be due to noisy ground truth supervision. I recommend to monitor the magnitude of the gradients in the network and the magnitude of the score matrix. Please close this issue if your problem is solved.

**Comment by atztao** [comment]
Thank you very much.
